Python/bytecodes.c

```diff
@@ -1584,6 +1600,19 @@ dummy_func(
             }
         }
 
+        inst(LOAD_SUPER_ATTR_METHOD, (unused/1, class_version/2, self_type_version/2, method/4, global_super, class, self -- res2, res)) {
+            DEOPT_IF(global_super != (PyObject *)&PySuper_Type, LOAD_SUPER_ATTR);
+            DEOPT_IF(!PyType_Check(class), LOAD_SUPER_ATTR);
+            DEOPT_IF(((PyTypeObject *)class)->tp_version_tag != class_version, LOAD_SUPER_ATTR);
+            PyTypeObject *self_type = Py_TYPE(self);
+            DEOPT_IF(self_type->tp_version_tag != self_type_version, LOAD_SUPER_ATTR);
```

Comment:
Given we expect a known self class and enclosing class, we also expect them to have a known relation.
So we expect `class` to be at a known index in `self.__class__`'s MRO.
We could (if we are seeking to shrink the cache) store the expected index into the MRO and check that (a single byte would be enough).

Probably would help performance though, as the additional indirection through the MRO wouldn't help.

Reply:
We discussed this in person. It seems unlikely to improve perf, would be a bit harder to understand the correctness, and probably wouldn't even save much cache size due to alignment? So we can keep it in our back pocket as a future idea for cache shrinking, but we'll leave it as is for now.